content/blog/intro-to-step-functions/index.md

```diff
@@ -0,0 +1,178 @@
+---
+title: "Intro to AWS Serverless Step Functions"
+date: 2020-02-13
+meta_desc: "An Introduction to building serverless apps with AWS Step Functions."
+meta_image: meta-step-function.png
+authors:
+    - sophia-parafina
+tags:
+    - serverless
+    - "step functions"
+---
+
+AWS Step Functions let you build applications by connecting AWS services. We'll go into the details of creating a lambda function, IAM roles and policies, and creating a workflow. Once we have the example deployed, we'll walk through the process of adding another function and step to the workflow. Included in the walkthrough is a discussion of one of the aspects of the Pulumi programming model. The goal of this article is to provide a foundation for building your application using serverless workflows.
+
+<!--more-->
+
+## Just the basics
+
+Let’s start with a hello world function in an AWS Lambda. You can start the [example from Github](https://github.com/pulumi/examples/tree/master/aws-py-stepfunctions") by clicking on the `Deploy with Pulumi` button. A new project page is created automatically in Pulumi console, click on `Create Project` to get started. Follow the steps to copy the project locally. As with any Python project, we recommend creating a virtual environment to store project-specific dependencies.
+
+```bash
+$ python3 -m venv venv
+$ source venv/bin/activate
+$ pip3 install -r requirements.txt
+```
+
+Before we run the example, let’s look at the code. A step function is made up of a lambda function and a state machine that runs the function. Both the lambda and the state machine require an [IAM execution role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html) and [policy](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies.html) to access AWS services and resources. The `iam.py` module establishes the roles and policies and the AWS region, which are imported in `__main__,py`.
+
+The lambda returns “Hello world!”  when called.  Note that the handler is the name of the module, `hello`, and the name of the function, `handler`.
+
+```python
+hello_world_fn = lambda_.Function('helloWorldFunction',
+   role=iam.lambda_role.arn,
+   runtime="python3.7",
+   handler="hello.handler",
+   code=pulumi.AssetArchive({
+       '.': pulumi.FileArchive('./step_hello')
```

Comment:
Is this right? The file is named `step_hello`? Should it be `step_hello.py` or something?

Reply:
It's pointing to the directory